### PR TITLE
feat(kit): `TextArea` extra characters highlight

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/demo/src/modules/components/text-area/text-area.template.html [hunk 1]
```diff
@@ -21,13 +21,22 @@
         </tui-doc-example>
 
         <tui-doc-example
-            id="large"
+            id="label-outside"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/text-area/text-area.template.html:24]: repeated id

File: projects/kit/components/text-area/text-area.component.ts [hunk 2]
```diff
@@ -139,6 +139,12 @@ export class TuiTextAreaComponent
         );
     }
 
+    get pseudoContent(): {fitted: string; extra: string} {
```
Comment (waterplea) [inline projects/kit/components/text-area/text-area.component.ts:142]: We do not create objects in getters, it has an overhead which is bad for performance. Make 2 getters instead.

File: projects/kit/components/text-area/text-area.style.less [hunk 3]
```diff
@@ -120,7 +120,7 @@
     white-space: pre-wrap;
     word-wrap: break-word;
     pointer-events: none;
-    visibility: hidden;
+    color: rgba(0, 0, 0, 0);
```
Comment (waterplea) [inline projects/kit/components/text-area/text-area.style.less:123]: transparent

File: projects/kit/components/text-area/text-area.style.less [hunk 4]
```diff
@@ -135,6 +135,10 @@
     :host._has-value._has-tooltip & {
         margin-right: 2.5rem;
     }
+
+    &__extra {
+        background-color: var(--tui-error-bg-night);
```
Comment (waterplea) [inline projects/kit/components/text-area/text-area.style.less:140]: Why night? Shouldn't it be --tui-error-bg by default and night for tuiMode onDark?

File: projects/kit/components/text-area/text-area.template.html [hunk 5]
```diff
@@ -45,12 +45,24 @@
...
-                    <div class="pseudo-content">{{value}}&nbsp;</div>
+                    <div
+                        class="pseudo-content"
+                        *tuiLet="pseudoContent as content"
+                        aria-hidden="true"
+                    >
+                        <span>{{content.fitted}}</span>
+                        <span
+                            class="pseudo-content__extra"
+                            *ngIf="content.extra"
```
Comment (waterplea) [inline projects/kit/components/text-area/text-area.template.html:56]: I believe there's no need for ngIf. It has no padding so empty span would already have 0 width and be invisible.

**Review by waterplea — CHANGES_REQUESTED** [review]
A few comments

File: projects/kit/components/text-area/text-area.style.less (hunk 4 above)
Comment (nsbarsukov) [reply projects/kit/components/text-area/text-area.style.less:140]: In Figma templates it is `--tui-error-bg-night` for default theme.
error-bg-night

File: projects/kit/components/text-area/text-area.component.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/kit/components/text-area/text-area.component.ts]: i made 2 getters

File: projects/kit/components/text-area/text-area.style.less (hunk 3 above)
Comment (nsbarsukov) [reply projects/kit/components/text-area/text-area.style.less]: fixed

File: projects/kit/components/text-area/text-area.template.html (hunk 5 above)
Comment (nsbarsukov) [reply projects/kit/components/text-area/text-area.template.html]: remove `*ngIf`

File: projects/kit/components/text-area/text-area.style.less (hunk 4 above)
Comment (waterplea) [reply projects/kit/components/text-area/text-area.style.less:140]: Ok, let's keep it that way

**Review by waterplea — APPROVED** [review]

**Pull request merged**